server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1996,6 +1997,29 @@ private RemoteSegmentStoreDirectory getRemoteDirectory() {
         return ((RemoteSegmentStoreDirectory) remoteDirectory);
     }
 
+    /**
+    Returns true iff it is able to verify that there is at least one
+    remote segment metadata uploaded
+     */
+    boolean isRemoteSync() {
+        assert indexSettings.isRemoteStoreEnabled();
+        try {
+            RemoteSegmentStoreDirectory directory = getRemoteDirectory();
+            if (directory.readLatestMetadataFile() != null) {
+                // verifying that all files except EXCLUDE_FILES are uploaded to the remote
+                Collection<String> uploadFiles = directory.getSegmentsUploadedToRemoteStore().keySet();
+                SegmentInfos segmentInfos = store.readLastCommittedSegmentsInfo();
+                Collection<String> localFiles = segmentInfos.files(true);
+                if (uploadFiles.containsAll(localFiles)) {
+                    return true;
```

Comment:
We could strengthen the checks by adding checksum comparisons for each file rather than just going with the names.

Reply:
Yes , we can , but we have our integrity checks at both upload and download flows. This is done right after upload , so doing that wouldn't add any benefit .